test/instances/morpho-blue/MorphoBlue_wstETH_USDC_Hyperdrive.t.sol

```diff
@@ -0,0 +1,818 @@
+                MarketParams({
+                    loanToken: LOAN_TOKEN,
+                    collateralToken: COLLATERAL_TOKEN,
+                    oracle: ORACLE,
+                    irm: IRM,
+                    lltv: LLTV
+                })
+            );
+        assertApproxEqAbs(totalSupplyAssets, totalBaseBefore + amountPaid, 1);
+        assertApproxEqAbs(
+            totalSupplyShares,
+            totalSharesBefore + hyperdrive.convertToShares(amountPaid),
+            1
+        );
+
+        // Ensure that the ETH balances didn't change.
+        assertEq(
+            address(hyperdrive).balance,
+            hyperdriveBalancesBefore.ETHBalance
+        );
+        assertEq(bob.balance, traderBalancesBefore.ETHBalance);
+
+        // Ensure that the Hyperdrive instance's base balance doesn't change
+        // and that the trader's base balance decreased by the amount paid.
+        assertEq(
+            IERC20(LOAN_TOKEN).balanceOf(address(hyperdrive)),
+            hyperdriveBalancesBefore.baseBalance
+        );
+        assertEq(
+            IERC20(LOAN_TOKEN).balanceOf(trader),
+            traderBalancesBefore.baseBalance - amountPaid
+        );
+
+        // Ensure that the shares balances were updated correctly.
+        (, uint256 hyperdriveSharesAfter) = getTokenBalances(
+            address(hyperdrive)
+        );
+        (, uint256 traderSharesAfter) = getTokenBalances(address(trader));
+        assertApproxEqAbs(
+            hyperdriveSharesAfter,
+            hyperdriveBalancesBefore.sharesBalance +
+                hyperdrive.convertToShares(amountPaid),
+            2
+        );
+        assertEq(traderSharesAfter, traderBalancesBefore.sharesBalance);
+    }
+
+    /// @dev Verifies that withdrawal accounting is correct when closing positions.
+    function verifyWithdrawal(
+        address trader,
+        uint256 baseProceeds,
+        bool asBase,
+        uint256 totalBaseBefore,
+        uint256 totalSharesBefore,
+        AccountBalances memory traderBalancesBefore,
+        AccountBalances memory hyperdriveBalancesBefore
+    ) internal view override {
+        // Vault shares withdrawals are not supported for this instance.
+        if (!asBase) {
+            revert IHyperdrive.UnsupportedToken();
+        }
+
+        // Ensure that the total supply decreased by the base proceeds.
+        (uint256 totalSupplyAssets, uint256 totalSupplyShares, , ) = MORPHO
+            .expectedMarketBalances(
+                MarketParams({
+                    loanToken: LOAN_TOKEN,
+                    collateralToken: COLLATERAL_TOKEN,
+                    oracle: ORACLE,
+                    irm: IRM,
+                    lltv: LLTV
+                })
+            );
+        assertApproxEqAbs(totalSupplyAssets, totalBaseBefore - baseProceeds, 1);
+        assertApproxEqAbs(
+            totalSupplyShares,
+            totalSharesBefore - hyperdrive.convertToShares(baseProceeds),
+            1e6
```

Comment:
Is this tolerance higher due to imprecision with convertToShares?

Reply:
I was able to drop this to 10.